src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/COE.cpp

```diff
@@ -641,6 +642,159 @@ COE COE::FromSIVector(const Vector6d& aCOEVector, const AnomalyType& anAnomalyTy
+    const Angle& aRaan,
+    const Angle& anAop,
+    const Angle& aTrueAnomaly
+)
+{
+    if ((aCelestialObjectSPtr == nullptr) || (!aCelestialObjectSPtr->isDefined()))
+    {
+        throw ostk::core::error::runtime::Undefined("Celestial object");
+    }
+
+    const Length equatorialRadius = aCelestialObjectSPtr->getEquatorialRadius();
+    const Real j2 = aCelestialObjectSPtr->accessGravitationalModel()->getParameters().J2_;
+    const Real j3 = aCelestialObjectSPtr->accessGravitationalModel()->getParameters().J3_;
+
+    return COE::FrozenOrbit(
+        aSemiMajorAxis, equatorialRadius, j2, j3, anEccentricity, anInclination, aRaan, anAop, aTrueAnomaly
+    );
+}
+
+COE COE::FrozenOrbit(
+    const Length& aSemiMajorAxis,
+    const Length& anEquatorialRadius,
+    const Real& aJ2,
+    const Real& aJ3,
+    const Real& anEccentricity,
+    const Angle& anInclination,
+    const Angle& aRaan,
+    const Angle& anAop,
+    const Angle& aTrueAnomaly
+)
+{
+    if (!aSemiMajorAxis.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Semi-major Axis");
+    }
+    if (!anEquatorialRadius.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Equatorial Radius");
+    }
+    if (!aJ2.isDefined() || aJ2 == 0.0)
+    {
+        throw ostk::core::error::runtime::Undefined("J2");
+    }
+    if (!aJ3.isDefined() || aJ3 == 0.0)
+    {
+        throw ostk::core::error::runtime::Undefined("J3");
+    }
+    if (!aRaan.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Raan");
+    }
+    if (!aTrueAnomaly.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("True anomaly");
+    }
+    if (anEccentricity.isDefined() && anInclination.isDefined())
+    {
+        throw ostk::core::error::RuntimeError("Cannot define both eccentricity and inclination");
+    }
+
+    const Array<Angle> criticalInclinations = {Angle::Degrees(63.4349), Angle::Degrees(116.5651)};
+    const Array<Angle> criticalAops = {Angle::Degrees(90.0), Angle::Degrees(270.0)};
+
+    const Real equatorialRadius_meters = anEquatorialRadius.inMeters();
+    const Real j2 = aJ2;
+    const Real j3 = aJ3;
+
+    // ecc =~ eccCoefficient * sin(incl)
+    const Real eccCoefficient = -j3 * equatorialRadius_meters / 2.0 / j2 / aSemiMajorAxis.inMeters();
+
+    const auto eccentricityFromInclination = [&eccCoefficient](const Angle& inclination) -> Real
+    {
+        return eccCoefficient * std::sin(inclination.inRadians());
+    };
+
+    const auto inclinationFromEccentricity = [&eccCoefficient](const Real& eccentricity) -> Angle
+    {
+        return Angle::Radians(asin(eccentricity / eccCoefficient));
+    };
+
+    // TBI: this only works because neither set of critical angles are 0/360 degrees
+    const auto isCritical = [](const Angle& angle, const Array<Angle>& criticalAngles) -> bool
+    {
+        const Real epsilon = Angle::Arcseconds(1.0).inRadians();  // TBI: make configurable?
```

Comment:
Indeed I do think the "criticality" angle tolerance should be configurable as an extra argument - preferably on the sooner side, because in reality if you area within a certain tolerance (I think much bigger than 1 Arcsec) of the conditions, your AOP and ecc will oscillate around the condition, instead of directly diverging - so it may actually be counterproductive to try and always correct and send us back to the perfect condition